scripts/update-doc-yml.py

```diff
@@ -0,0 +1,38 @@
+"""
+Author: Jack He (ProjectsByJackHe)
+
+This utility script is useful for creating a toc.yml file specifically for use with DocFx.
+
+DocFx is a dotnet tool for generating documentation from markdown files.
+It requires a toc.yml file to link markdowns to generate the static site.
+"""
+
+import os
```

Comment:
IMO, generating toc.yml should be part of the github workflow. Is it really necessary to store toc.yml files in the repo (vs generating them and publishing them to GitHub pages)?

Reply:
That's a good point! 

According to the official DocFx Github (https://github.com/dotnet/docfx/tree/main/docs) we're actually not supposed to auto-generate toc.yml as that is our manifest for generating the site, and it's meant to be manually updated, but that's fine since it's isn't often we add / remove markdowns. 

With that said, I didn't want to type out manually the gigantic amount of markdown files we have, so I wrote a Python script for it. But in the future, we can certainly setup something in the build process to call my python / powershell script to dynamically create the toc.yml file, and have it ready just in time to build the full static site.

Reply:
Ok, I updated the powershell script to recursively generate a toc.yml for any subdirectories, so we could run that in the github workflow to create the toc.yml just in time in the github runner.